packages/store/src/ivy/ivy-enabled-in-dev-mode.ts

```diff
@@ -1,30 +1,48 @@
-import { isDevMode } from '@angular/core';
 import { ReplaySubject } from 'rxjs';
 
-export const ivyEnabledInDevMode$ = new ReplaySubject<boolean>(1);
+import { getUndecoratedStateInIvyWarningMessage } from '../configs/messages.config';
+
+// We've got such expression because Terser doesn't tree-shake `new SomeClass()` expressions
+// since Terser considers them as "side-effectful". E.g. `new InjectionToken()` expression isn't
+// tree-shakable even if the token has `providedIn: root` and is not used.
+// The `ReplaySubject` will be subscribed inside the `ngDevMode` guard anyway.
+export const ivyEnabledInDevMode$ =
+  // Caretaker note: we have still left the `typeof` condition in order to avoid
+  // creating a breaking change for projects that still use the View Engine.
+  typeof ngDevMode === 'undefined' || ngDevMode ? new ReplaySubject<boolean>(1) : null!;
```

Comment:
This will be tree-shaken if Ivy is enabled since `false ? new ReplaySubject() : null` will be transformed into `const ivyEnabledInDevMode$ = null`. Terser will be able to eliminate this expression. Otherwise it will behave in the same way as before for VE projects.

Reply:
It feels a bit weird to set it to null because of the functions beneath that use it.
If we don't set this to null, then would this still possibly be tree shaken because the functions below are treeshaken?

Reply:
Sorry, I’m a bit of unaware of the context since this seems obvious for me.

It will be set to `null` when ngDevMode is false and the function beneath will use it when it doesn’t equal null. So basically there will be no exception and as you see all tests are passing.

There would be no reason to set it to null but we do that because of `new ReplaySubject`, since _new_ expressions are not tree-shakable (even if they are not referenced and used).

Reply:
You are awesome! I learned something from you today.